crates/services/p2p/src/peer_manager.rs

```diff
@@ -645,28 +636,33 @@ impl PeerManager {
     fn handle_peer_disconnect(&mut self, peer_id: PeerId) {
         // try immediate reconnect if it's a reserved peer
         let is_reserved = self.reserved_peers.contains(&peer_id);
+        let is_removed;
```

Comment:
In the `handle_initial_connection` we don't connect to the peer -> it absents in our `self.non_reserved_connected_peers` map.

We had a bug here that we will `allow_new_peers` even if the peer was not connected -> we allowed new connections when we had `max_non_reserved_peers`.